File: App/FantasyGolf/src/main/java/Player.java
public class Player {
    private String name;
    private int ranking = -1;
    private boolean first;
    private boolean second;
    private int finish;
    private boolean madeCut;
    private double points;

    // A player is created. Entering name, ranking, true/false
    // if first choice. false/true if second. false/fase if neither
    public Player(String name, boolean first,
            boolean second) {
        this.name = name;
        if (first && second) {
            throw new IllegalArgumentException("Can't be first and second choice!");
        }

        this.first = first;
        this.second = second;
    }

    public void inputResults(int finish, boolean madeCut) {
        this.finish = finish;
        this.madeCut = madeCut;
    }

    public void setRanking(int ranking) {
        this.ranking = ranking;
    }

    public void score() {
        double currentPoints = 0;

        // 5 points if made cut and ranking outside top 5
        if (madeCut && ranking > 5) {
            currentPoints += 5;
        }
        if (finish <= 25) {
            currentPoints += 3;
            // if ranking outside of top 10 but in top 20, 6 bonus points
            if (ranking <= 20 && ranking > 10) {
                currentPoints += 6;
            }
            if (ranking > 20) {
                currentPoints += 11;
            }
        }

        // 4 points to top 15
        if (finish <= 15) {
            currentPoints += 4;
        }
        if (finish <= 10) {
            currentPoints += (11 - finish);
        }

        // 15 bonus points if winner
        if (finish == 1) {
            currentPoints += 15;
        }

        // First and Second choice players bonus
        if (first) {
            currentPoints = currentPoints * 2;
        }
        if (second) {
            currentPoints = currentPoints * 1.5;
        }

        this.points = currentPoints;
    }

    public int getRanking() {
        return this.ranking;
    }

    public boolean getCut() {
        return this.madeCut;
    }

    public String getName() {
        return this.name;
    }

    /**
     * 
     * @return the amount of points a player scored
     */
    public double getPoints() {
        return this.points;
    }

    public int getFinish() {
        return finish;
    }
}

File: App/FantasyGolf/main/player.py
class Player:
    
    def __init__(self, name, ranking, first, second, finish, status, worst_top_25):
        self.name = name
        self.ranking = ranking
        self.first = first
        self.second = second
        self.finish = finish 
        if status == "cut" or status == "withdrawn":
            self.made_cut = False
        else:
            self.made_cut = True

        self.score = 0.0
        self.worst_top_25 = worst_top_25

    def calculate_score(self, worst_in_top_25):
        '''
        Calculates the total points this player got (not included bonus for first/second pick)
        '''
        # points for making the cut if ranked > 5
        if self.made_cut and self.ranking >= 5:
            self.score += 5

        #top 25
        if self.finish <= 25:
            self.score += 3
            if self.ranking <= 20 and self.ranking > 10:
                self.score += 6
            if self.ranking > 20:
                self.score += 11
        
        #top 15
        if self.finish <= 15:
            self.score += 4
        
        #top 10
        if self.finish <= 10:
            self.score += (11 - self.finish)
        
        if self.finish == 1:
            self.score += 15

        if self.name == worst_in_top_25:
            self.worst_top_25 = True
        
        if self.first:
            self.score = self.score*2
        
        if self.second:
            self.score = self.score * 1.5


    def get_score(self):
        return self.score

    def get_made_cut(self):
        return self.made_cut

    def JSONify(self):
        obj = {}
        obj["Name"] = self.name
        obj["Cut"] = not self.made_cut
        obj["Points scored"] = self.score
        obj["Finish"] = int(self.finish)
        return obj
    
    def __str__(self):
        return f"Player {self.name}({self.ranking}), position: {self.finish} score: {self.score}"
        
        


